GitHub Action Azure Federated Identity (#72)
mspnp/aks-baseline-multi-region: docs/deploy/06-aks-cluster.md

    echo RESOURCEID_VNET_BU0001A0042_04: $RESOURCEID_VNET_BU0001A0042_04
    ```

    1. Create the Azure Credentials for the [GitHub CD workflow](https://learn.microsoft.com/azure/developer/github/connect-from-azure).
        :warning: Set your GitHub Account

        ```bash
        GITHUB_ACCOUNT_NAME=YOUR-GITHUB-ACCOUNT-NAME-GOES-HERE

        # Creating a User Managed Identity which will be Federated to GitHub Action

>         az deployment group create -g rg-bu0001a0042-shared -f ghaction-federated-identity.bicep -p gitAccount="$GITHUB_ACCOUNT_NAME"

Reviewer: Can you instead just add this to step 3's bicep file?  Why make this something else?  Then it will already be in place for this step.

Also, has already been captured as `$GITHUB_USER_NAME_AKS_MRB` so no need to get it again.  So you can provide that as a parameter in step 3's deployment.

Author: I didn't add to other deploy file because this is a resource which is only needed because we are using gitHubAction, it is not part of the aks baseline narrative. I thought that it would be better keep isolated.   
We can move to az cli instructions or move some pre-existent deploy file (This reference implementation still use arm, it is the first bicep).   
Good catch about the variable, it was changed https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/7e97c8836b20dfcf1244ac0eea6f46ba075cbd61

Reviewer: Thanks for your comment on teams, let's go ahead and just add it to the shared json ARM file.  When you're porting it over, I'd suggest a parameter name of `gitHubAccountName`.  That'll match our variable name and will disambiguate between a git username and a GitHub account.

Author: https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/5438bbbc093577ad9f260a073e0b7ddcdde81d86

>         export APP_ID=$(az deployment group show -g rg-bu0001a0042-shared -n ghaction-federated-identity --query 'properties.outputs.clientId.value' -o tsv)

Reviewer: Suggestion: Be more clear instead of `APP_ID` give it a meaningful name like `GitHubFederatedIdentity_ClientID`

Author: done https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/7e97c8836b20dfcf1244ac0eea6f46ba075cbd61

Author: https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/5438bbbc093577ad9f260a073e0b7ddcdde81d86

    echo RESOURCEID_VNET_BU0001A0042_04: $RESOURCEID_VNET_BU0001A0042_04
    ```

>     1. Assign standard permission on the subccription for the identity which will use the [GitHub CD workflow](https://learn.microsoft.com/azure/developer/github/connect-from-azure).

Reviewer: ```suggestion
    1. Assign required permissions to the [GitHub workflow's](https://learn.microsoft.com/azure/developer/github/connect-from-azure) managed (federated) identity.
```

        ```bash

>         # Getting the deployed identity to assign standard permissions

Reviewer: ```suggestion
        # Get the managed identity to assign necessary deployment permissions
```

        export GITHUB_FEDERATED_IDENTITY_CLIENTID=$(az deployment group show -g rg-bu0001a0042-shared -n shared-svcs-stamp --query 'properties.outputs.githubFederatedIdentityClientId.value' -o tsv)
        echo GITHUB_FEDERATED_IDENTITY_CLIENTID: $GITHUB_FEDERATED_IDENTITY_CLIENTID

        # Assign built-in Contributor RBAC role for creating resource groups and performing deployments at subscription level
        az role assignment create --assignee $GITHUB_FEDERATED_IDENTITY_CLIENTID --role 'Contributor'

        # Assign built-in User Access Administrator RBAC role since granting RBAC access to other resources during the cluster creation will be required at subscription level (e.g. AKS-managed Internal Load Balancer, ACR, Managed Identities, etc.)
        az role assignment create --assignee $GITHUB_FEDERATED_IDENTITY_CLIENTID --role 'User Access Administrator'
        ```

>     1. Create secrets in your GitHub repository.

Reviewer: ```suggestion
    1. Create federated identity secrets in your GitHub repository.
```

>         > :bulb: Use the content from the `sp.json` file.

Reviewer: what sp.json?

Author: This line needs to be deleted. Good Point.

Author: https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/0cf85dcb8cb79aa133564b7cd623db9f464e24d5

        az role assignment create --assignee $APP_ID --role 'User Access Administrator'
        ```

    1. Create secrets in your GitHub repository.

        > :bulb: Use the content from the `sp.json` file.

        ```bash
        export TENANT_ID=$(az deployment group show -g rg-bu0001a0042-shared -n ghaction-federated-identity --query 'properties.outputs.tenantId.value' -o tsv)
        echo $TENANT_ID
        export SUBSCRIPTION_ID=$(az account show | jq -r '.id')

>         echo SUBSCRIPTION_ID: $SUBSCRIPTION_ID

Reviewer: `jq` should not be a dependency.  Otherwise we need to list that on the dependency page, and that would be a pretty big hammer for such little usage.  Please use `--query` instead.

Author: done https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/7e97c8836b20dfcf1244ac0eea6f46ba075cbd61

>         gh secret set AZURE_CLIENT_ID -b"${APP_ID}"  --repo $GITHUB_ACCOUNT_NAME/aks-baseline-multi-region

Reviewer: Use the already existing `$GITHUB_USER_NAME_AKS_MRB` for these repo names here and below.  And nice touch adding that in to be explicit!

Author: done https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/7e97c8836b20dfcf1244ac0eea6f46ba075cbd61

Reviewer: You still have ":warning: Set your GitHub Account" in there.

Author: https://github.com/mspnp/aks-baseline-multi-region/pull/72/commits/5438bbbc093577ad9f260a073e0b7ddcdde81d86
